src/main/java/jaryl/duke/Output.java

```diff
@@ -0,0 +1,207 @@
+        else
+            return "Noted. I've marked this task as undone: \n\t" + task;
+    }
+
+    /**
+     * Prints message upon successfully deleting a task
+     * @param task      the task to be deleted
+     * @param numTasks  total number of existing tasks
+     */
+    public String printDeleteMsg(Task task, int numTasks) {
+        return "Noted. I've removed this task: \n\t" + task + "\nNow you have " + numTasks + " tasks in the list.";
+    }
+
+    /**
+     * Prints message for user query
+     */
+    public String printFind(ArrayList<Task> foundTasks, String query) {
+        StringBuilder sb = new StringBuilder();
+        if (foundTasks.size() == 0) {
+            return "☹ OOPS! No tasks found for the query: \n\t" + query;
+        }
+        for (Task t : foundTasks) {
+            sb.append(t);
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Prints message when user inputs an illegal argument
+     */
+    public String printIllegalArgumentError() {
+        return "☹ OOPS! I'm sorry, but I don't know what that means :(";
+    }
+
+    /**
+     * Handles list command
+     * @param tasksList list of tasks
+     */
+    public String listAction(ArrayList<Task> tasksList) throws DukeException {
+        StringBuilder sb = new StringBuilder();
+        if(tasksList.size() == 0) {
+            throw new EmptyListException();
+        }
+
+        for(int i = 1; i <= tasksList.size(); i++) {
+            Task task = tasksList.get(i - 1);
+            sb.append(i + ". " + task + "\n");
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * Handles done command
+     * @param tasksList     list of tasks
+     * @param input         user input
+     * @param dataManager   data manager which handles reading/writing data
+     */
+    public String doneAction(ArrayList<Task> tasksList, String input, DataManager dataManager) throws DukeException {
+        String[] params = input.split(" ");
+
+        if (params.length < 2) {
+            throw new InvalidFormatException("Invalid event number. Please specify a valid event you would like to mark done");
+        } else if (Integer.parseInt(params[1]) <= 0 || Integer.parseInt(params[1]) > tasksList.size()) {
+            throw new InvalidFormatException("Invalid event number. Please specify a valid event you would like to mark done");
+        }
+        Task done = tasksList.get(Integer.parseInt(params[1]) - 1);
+        done.toggleStatus();
+
+        dataManager.writeToFile(tasksList);
+        return(printDoneMsg(done));
+    }
+
+    /**
+     * Handles todo, deadline and event commands
+     * @param tasksList     list of tasks
+     * @param input         user input
+     * @param dataManager   data manager which handles reading/writing data
+     */
+    public String addAction(ArrayList<Task> tasksList, String input, DataManager dataManager) throws DukeException {
+        String taskDesc = "", dateTime = "";
+        String[] checkFormat;
```

Comment:
Should variables which represent a collection of values be named in the plural sense?